browser/brave_profile_prefs.cc

```diff
@@ -409,6 +409,8 @@ void RegisterProfilePrefs(user_prefs::PrefRegistrySyncable* registry) {
 
   registry->SetDefaultPrefValue(prefs::kEnableMediaRouter, base::Value(false));
 
+  registry->RegisterBooleanPref(kBraveMediaRouter, false);
```

Comment:
I don't understand why we're adding a new pref and how is this different from `prefs::kEnableMediaRouter`?

Reply:
Cast extension requires a browser restart once the setting is toggled. We use the pref `kBraveMediaRouter` as a proxy to identify the current state of the switch and `prefs::kEnableMediaRouter` is updated to `kBraveMediaRouter` on browser start.

Reply:
I thought we already had something like this?